edwin/client/static/js/components/Timeline.js

```diff
@@ -82,6 +99,22 @@ Timeline.propTypes = {
   params: React.PropTypes.object.isRequired,
 };
 
+class Indicator extends React.Component {
+  render() {
+    let state = this.props.state.toJS().state;
```

Comment:
Oh, I don't mind this being here. I just think it should be named something besides `state`. Maybe `progressState` or something. It's like naming something `filter` in Python. Yes, it works fine, but it can be confusing. (though it isn't shadowing anything here)

As far as the display/comparison thing: Display is annoying, yes. As far as comparison, the constants that come out of `constantMap` are actually just strings. I think the part you are having trouble with is probably the Immutable data structures, which generally display as something entirely unhelpful.